File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/dtos/ProblemDto.java
package io.github.sanyavertolet.edukate.backend.dtos;

import io.github.sanyavertolet.edukate.backend.entities.Problem;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.springframework.data.annotation.PersistenceCreator;

import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor(onConstructor = @__(@PersistenceCreator))
@RequiredArgsConstructor
public class ProblemDto {
    private String id;
    private Boolean isHard;
    private List<String> tags;
    private String text;
    private List<Problem.Subtask> subtasks = new ArrayList<>();
    private List<String> images;
    private Boolean hasResult;
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/dtos/ProblemDto.kt
package io.github.sanyavertolet.edukate.backend.dtos

import io.github.sanyavertolet.edukate.backend.entities.Problem
import io.github.sanyavertolet.edukate.backend.entities.Subproblem
import io.github.sanyavertolet.edukate.common.ContentLanguage

data class ProblemDto(
    val key: String,
    val code: String,
    val bookSlug: String,
    val isHard: Boolean,
    val tags: List<String>,
    val text: String,
    val subproblems: List<Subproblem>,
    val images: List<String>,
    val status: Problem.Status,
    val hasResult: Boolean,
    val language: ContentLanguage,
)
